### PR TITLE
Allow to choose the kind of optimization

### PR DESCRIPTION
**Pull request by shargon**
While we are testing the new optimization, we should be able to disable it.

### CODE REVIEW

File: src/Neo.Compiler.CSharp/CompilationOptions.cs [hunk 1]
```diff
@@ -10,15 +10,26 @@
...
 namespace Neo.Compiler
 {
     public class CompilationOptions
     {
+        [Flags]
+        public enum OptimizationType : byte
+        {
+            None = 0,
+            Basic = 1,
+            Experimental = 2,
```
Comment (vncoelho) [inline src/Neo.Compiler.CSharp/CompilationOptions.cs:25]: Expiremental in the release as well?

Comment (shargon) [reply src/Neo.Compiler.CSharp/CompilationOptions.cs:25]: Yes

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/CompilationOptions.cs:25]: A different name please, too long~~

Comment (shargon) [reply src/Neo.Compiler.CSharp/CompilationOptions.cs:25]: Ideas? people use experimental always

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/CompilationOptions.cs:25]: o0, o1, o2, o3

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/CompilationOptions.cs:25]: I think this is commonly used by compilers.

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/CompilationOptions.cs:25]: I don't think it's a good thing to ask user type a long word. But I am not going to ask you to do it.

**Review by Jim8y — APPROVED** [review]

File: src/Neo.Compiler.CSharp/CompilationOptions.cs (hunk 1 above)
Comment (shargon) [reply src/Neo.Compiler.CSharp/CompilationOptions.cs:25]: I don't think that no one should use only experimental, just use `all`

**Pull request merged**